aspnetcore/blazor/fundamentals/environments.md

```diff
@@ -135,57 +152,150 @@ In the following example for IIS, the custom header (`Blazor-Environment`) is ad
 
-## Read the environment
+## Read the environment in a Blazor WebAssembly app
 
 Obtain the app's environment in a component by injecting <xref:Microsoft.AspNetCore.Components.WebAssembly.Hosting.IWebAssemblyHostEnvironment> and reading the <xref:Microsoft.AspNetCore.Components.WebAssembly.Hosting.IWebAssemblyHostEnvironment.Environment> property.
 
 `ReadEnvironment.razor`:
 
-<!-- UPDATE 8.0 Watch the highlights! -->
+```razor
+@page "/read-environment"
+@using Microsoft.AspNetCore.Components.WebAssembly.Hosting
+@inject IWebAssemblyHostEnvironment Env
 
-:::moniker range=">= aspnetcore-7.0"
+<h1>Environment example</h1>
 
-:::code language="razor" source="~/../blazor-samples/7.0/BlazorSample_WebAssembly/Pages/environments/ReadEnvironment.razor" highlight="3,7":::
+<p>Environment: @HostEnvironment.Environment</p>
+```
 
-:::moniker-end
+:::moniker range=">= aspnetcore-8.0"
 
-:::moniker range=">= aspnetcore-6.0 < aspnetcore-7.0"
+## Read the environment in a Blazor Web App
 
-:::code language="razor" source="~/../blazor-samples/6.0/BlazorSample_WebAssembly/Pages/environments/ReadEnvironment.razor" highlight="3,7":::
+Assuming that prerendering isn't disabled for a component or the app, a component in the `.Client` project is prerendered on the server. Because the server doesn't have a registered <xref:Microsoft.AspNetCore.Components.WebAssembly.Hosting.IWebAssemblyHostEnvironment> service, it isn't possible to inject the service and use the service implementation's host environment extension methods and properties during server prerendering. Injecting the service into an Interactive WebAssembly or Interactive Auto component results in the following runtime error:
 
-:::moniker-end
+> :::no-loc text="There is no registered service of type 'Microsoft.AspNetCore.Components.WebAssembly.Hosting.IWebAssemblyHostEnvironment'.":::
 
-:::moniker range=">= aspnetcore-5.0 < aspnetcore-6.0"
+To address this, create a shared service abstraction and instantiate the service in the server and `.Client` projects with the host environment. The following example demonstrates the approach.
 
-:::code language="razor" source="~/../blazor-samples/5.0/BlazorSample_WebAssembly/Pages/environments/ReadEnvironment.razor" highlight="3,7":::
+In the client project (`.Client`):
 
-:::moniker-end
+* Add an interface for the service abstraction.
+* Add a service implementation that obtains the host environment from <xref:Microsoft.AspNetCore.Components.WebAssembly.Hosting.IWebAssemblyHostEnvironment.Environment%2A?displayProperty=nameWithType>.
 
-:::moniker range="< aspnetcore-5.0"
+`IHostEnvironment.cs`:
+
+```csharp
+public interface IHostEnvironment
+{
+    string EnvironmentName { get; set; }
+}
+```
+
+`WasmHostEnvironment.cs`:
+
+```csharp
+using Microsoft.AspNetCore.Components.WebAssembly.Hosting;
+
+public class WasmHostEnvironment(IWebAssemblyHostEnvironment env) : IHostEnvironment
+{
+    public string EnvironmentName { get; set; } = env.Environment;
```

Comment:
Could this be a computed property (using `=>` rather than `{ get; set; } =`)? Same suggestion for `ServerHostEnvironment`. I only suggest this because `IWebHostEnvironment` has a mutable `EnvironmentName` property, and we would want the `EnvironmentName` property here to reflect any changes to the underlying environment name.

Reply:
Yes ... again, it looked like at first that the compiler was going to choke on it. ***No!*** The compiler is good. I'll make the change. 👍